tests/agents/test_opencode_chat.py

```
@@ -0,0 +1,367 @@
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup_env(self):
+        base_url = os.environ.get("OPENCODE_BASE_URL", "http://localhost:4096")
+        self.base_url = base_url
+        self.password = os.environ.get("OPENCODE_SERVER_PASSWORD")
+
+    def test_health_check_returns_healthy(self):
+        import urllib.request
+
+        req = urllib.request.Request(f"{self.base_url}/global/health")
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)